### PR TITLE
[Amazon Q]: Emit user modification SendTelemetryEvent events for chat and inline completions

### PR DESCRIPTION
**Pull request by viktorsaws**
## Problem
Implement UserModification and ChatUserModification telemetry events for Amazon Q.

## Solution

* Added emitting `chatUserModificationEvent` hooking to existing mechanism for Toolkits telemetry
* Wire `CodeDiffTracker` into `CodeWhispererServer` inline completion handler, and emitting `UserModification` telemetry on timer.

Verified that telemetry is arriving to service backend.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: server/aws-lsp-codewhisperer/src/language-server/codeWhispererServer.ts [hunk 1]
```diff
@@ -324,6 +337,23 @@ export const CodewhispererServerFactory =
...
+        const codeDiffTracker: CodeDiffTracker<AcceptedInlineSuggestionEntry> = new CodeDiffTracker(
+            workspace,
+            logging,
+            (entry: AcceptedInlineSuggestionEntry, percentage, unmodifiedAcceptedCharacterCount) => {
+                telemetryService.emitUserModificationEvent({
+                    sessionId: entry.sessionId,
+                    requestId: entry.requestId,
+                    languageId: entry.languageId,
+                    customizationArn: entry.customizationArn,
+                    timestamp: new Date(),
```
Comment (pras0131) [inline server/aws-lsp-codewhisperer/src/language-server/codeWhispererServer.ts:349]: nit: for other events where we need to emit timestamp, we are calculating the same in TelemetryService. We can follow the same for this event?

Comment (pras0131) [reply server/aws-lsp-codewhisperer/src/language-server/codeWhispererServer.ts:349]: Also, do you mean to add date.now() here?

File: server/aws-lsp-codewhisperer/src/language-server/codeWhispererServer.ts [hunk 2]
```diff
@@ -534,6 +565,23 @@ export const CodewhispererServerFactory =
...
+        // Schedule tracker for UserModification Telemetry event
+        const enqueueCodeDiffEntry = (session: CodeWhispererSession, acceptedSuggestion: Suggestion) => {
+            const endPosition = getEndPositionForAcceptedSuggestion(acceptedSuggestion.content, session.startPosition)
+
+            codeDiffTracker.enqueue({
+                sessionId: session.codewhispererSessionId || '',
+                requestId: session.responseContext?.requestId || '',
+                fileUrl: session.document.uri,
+                languageId: session.language,
+                time: Date.now(),
```
Comment (pras0131) [inline server/aws-lsp-codewhisperer/src/language-server/codeWhispererServer.ts:577]: same as above, we can move population of timestamp to TelemetryService

File: server/aws-lsp-codewhisperer/src/language-server/codeWhispererServer.ts (hunk 1 above)
Comment (viktorsaws) [reply server/aws-lsp-codewhisperer/src/language-server/codeWhispererServer.ts:349]: I used `new Date()`, as it is in line with how it's done in VSCode Toolkit

**Review by pras0131 — APPROVED** [review]

**Pull request merged**